execute/window.go

```diff
@@ -18,10 +18,11 @@ type Window struct {
 // and normalizes the offset to a small positive duration.
 // It also validates that the durations are valid when
 // used within a window.
-func NewWindow(every, period, offset Duration) (Window, error) {
-	// Normalize the offset to a small positive duration
-	offset = offset.Normalize(every)
-
+func NewWindow(every, period, offset Duration, months bool) (Window, error) {
+	if !months {
+		// Normalize nanosecond offsets to a small positive duration
+		offset = offset.Normalize(every)
+	}
```

Comment:
@jlapacik we need to know if we're working with nanoseconds or not so that we cna know whether to scale durations here or not.

Reply:
I see so it wasn't a bug in `GetEarliestBounds`? Just in how we were creating the window in the first place? Is it possible to add that test case back in?

Reply:
Sure, I can add another case that's similar to the initial one.